src/Sulu/Bundle/AdminBundle/Exception/DeletionImpossibleChildPermissionsException.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\AdminBundle\Exception;
+
+class DeletionImpossibleChildPermissionsException extends \Exception
```

Comment:
I'm still unhappy with those exception class names, but I can't find a better one. Does somebody have an idea?

Reply:
Also not sure maybe @nnatter has an idea how we could call this exceptions.

Just something which came to my mind:

 - DenyRemoveByChildPermissionsException
 - ConflictRemoveByExistChildrenException

Reply:
what about those?

- `DeletionWithChildrenNotAllowedException`
- `InsufficientChildPermissionsException`

Reply:
havent had a good idea either 🙈 

i would prefer `InsufficientChildPermissionsException`/`MissingChildPermissionsException`, but i guess we need to include `Deletion` because of the message right?
maybe `DeletionBlockedByInsufficientChildPermissionsException`?

Reply:
No we don't need to make this exception specific for deletion. I guess we could also show this exception, if a user changes the permissions of a collection and checks `Override permissions of sub elements`, but doesn't have the `security` permission of one of the child collections.

Reply:
Additionally, we could also add a `permission` property, which can be used to determine, if the user tried to delete or modify